src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -220,4 +321,152 @@ mod tests {
+            } else {
+                evaluator = evaluator.push_param(arg1);
+            }
+
+            let output: Option<Int> = evaluator.evaluate(sig).unwrap();
+            assert_eq!(
+                output.unwrap() as u64,
+                expect,
+                "{:?}, {:?}, {:?}, {:?}, {:?}",
+                arg0,
+                arg0_unsigned,
+                arg1,
+                arg1_unsigned,
+                sig
+            );
+        }
+    }
+
+    #[test]
+    fn test_mod_real() {
+        let tests = vec![
+            (ModReal, Some(1.0), None, None),
+            (ModReal, None, Some(1.0), None),
+            (ModReal, Some(1.0), Some(1.1), Some(1.0)),
+            (ModReal, Some(-1.0), Some(1.1), Some(-1.0)),
+            (ModReal, Some(1.0), Some(-1.1), Some(1.0)),
+            (ModReal, Some(-1.0), Some(-1.1), Some(-1.0)),
+            (ModReal, Some(1.0), Some(0.0), None),
+        ];
+
+        for (sig, arg0, arg1, expect) in tests {
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg0)
+                .push_param(arg1)
+                .evaluate(sig)
+                .unwrap();
+            assert_eq!(output, expect, "{:?}, {:?}, {:?}", arg0, arg1, sig);
+        }
+    }
+
+    #[test]
+    fn test_mod_decimal() {
+        fn str2dec(str: &'static str) -> Option<Decimal> {
+            Some(str.parse().unwrap())
+        }
+
+        let tests = vec![
+            (ModDecimal, str2dec("13"), str2dec("11"), str2dec("2")),
+            (ModDecimal, str2dec("-13"), str2dec("11"), str2dec("-2")),
+            (ModDecimal, str2dec("13"), str2dec("-11"), str2dec("2")),
+            (ModDecimal, str2dec("-13"), str2dec("-11"), str2dec("-2")),
+            (ModDecimal, str2dec("33"), str2dec("11"), str2dec("0")),
+            (ModDecimal, str2dec("-33"), str2dec("11"), str2dec("0")),
+            (ModDecimal, str2dec("33"), str2dec("-11"), str2dec("0")),
+            (ModDecimal, str2dec("-33"), str2dec("-11"), str2dec("0")),
+            (
+                ModDecimal,
+                str2dec("0.0000000001"),
+                str2dec("1.0"),
+                str2dec("0.0000000001"),
+            ),
+            (ModDecimal, str2dec("1"), str2dec("1.1"), str2dec("1")),
+            (ModDecimal, str2dec("-1"), str2dec("1.1"), str2dec("-1")),
+            (ModDecimal, str2dec("1"), str2dec("-1.1"), str2dec("1")),
+            (ModDecimal, str2dec("-1"), str2dec("-1.1"), str2dec("-1")),
+            (ModDecimal, str2dec("3"), str2dec("0"), None),
+            (ModDecimal, str2dec("-3"), str2dec("0"), None),
+            (ModDecimal, str2dec("0"), str2dec("0"), None),
+            (ModDecimal, str2dec("-3"), None, None),
+            (ModDecimal, None, str2dec("-3"), None),
+            (ModDecimal, None, None, None),
+        ];
+
+        for (sig, arg0, arg1, expect) in tests {
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg0.clone())
```

Comment:
str2dec can be removed if we parse the string into decimal here like @sticnarf did in Plus decimal function :)